hw_diag/views/diagnostics.py

```diff
@@ -111,3 +114,46 @@ def version_information():
     }
 
     return response
+
+
+@DIAGNOSTICS.route('/v1/add-gateway-txn', methods=['POST'])
+def add_gateway_txn():
+    """
+    Generates an add_gateway_txn if a destination name and wallets are defined and valid.
+    Diagnostics report will be in the format below if successful.
+
+    https://docs.helium.com/mine-hnt/full-hotspots/become-a-maker/hotspot-integration-testing/#generate-an-add-hotspot-transaction
+    {
+        errors: [],
+        DESTINATION_ADD_GATEWAY_TXN_KEY: {
+            "address": "11TL62V8NYvSTXmV5CZCjaucskvNR1Fdar1Pg4Hzmzk5tk2JBac",
+            "fee": 65000,
+            "owner": "14GWyFj9FjLHzoN3aX7Tq7PL6fEg4dfWPY8CrK8b9S5ZrcKDz6S",
+            "payer": "138LbePH4r7hWPuTnK6HXVJ8ATM2QU71iVHzLTup1UbnPDvbxmr",
+            "staking fee": 4000000,
+            "txn": "CrkBCiEBrlImpYLbJ0z0hw5b4g9isRyPrgbXs9X+RrJ4pJJc9MkS..."
+        }
+    }
+    """
+
+    shipping_destination_with_signature = request.get_data()
+    if not shipping_destination_with_signature:
+        err_msg = 'Can not find payload.'
+        LOGGER.error(err_msg)
+        diagnostics_report = compose_diagnostics_report_from_err_msg(
+            DESTINATION_ADD_GATEWAY_TXN_KEY, err_msg)
+        return diagnostics_report, 406
+
+    diagnostics = [
+        AddGatewayTxnDiagnostic(shipping_destination_with_signature),
+    ]
+    diagnostics_report = DiagnosticsReport(diagnostics)
+    diagnostics_report.perform_diagnostics()
+    if diagnostics_report.has_errors({DESTINATION_WALLETS_KEY}):
+        http_code = 406
```

Comment:
That's usually when data provided by the client has errors. In our case if the signed payload does not get "decrypted" correctly.

If there were errors during gateway transaction generation, isn't that a server side issue and more worthy of a 5xx error?

Reply:
For the failure of add-gateway-txn, it can be considered as a server-side error, so I will return 500 in this case. Thank you.